int8 air_pump_close
int8 execute_finished
